lib/src/socket.dart

```diff
@@ -249,12 +268,14 @@ class Socket {
 
   // Return the next message ref, accounting for overflows
   String makeRef() {
-    final newRef = ref + 1;
-    if (newRef == ref) {
-      ref = 0;
+    const int int64MaxValue = 9223372036854775807;
+    int newRef;
+    if (ref == int64MaxValue) {
```

Comment:
I do a quick check `int64MaxValue + 10 = -9223372036854775799`
so it's safer to do 
```dart 
if (ref < 0 || ref >= int64MaxValue)
```
what do you think?

Reply:
In that case, I think we should do this:
```
int newRef = ref + 1;
if (newRef < 0)  {
    newRef = 0;
}
ref = newRef;
```

Reply:
https://github.com/dart-lang/sdk/blob/master/docs/language/informal/int64.md
> Integers wrap around, or worded differently, all operations are done modulo 2^64. If an operation overflows (that is the resulting value exceeds the maximum int64 value 9,223,372,036,854,775,807) then the value is reduced modulo 2^64 and the resulting bits are treated as a signed 64 bit integer (using 2's complement for negative numbers). A similar operation happens when the operation underflows.

yeah, your change looks good to me.